common_code/inspiral/CE_orbit.f90

```diff
@@ -242,17 +383,32 @@ subroutine calc_quantities_at_comp_position(id, ierr)
 
          M2 = CE_companion_mass * Msun
 
+         ! Determining the temperature at the position of the companion
+         temp = s% T(k)
+
          ! Determine orbital period in seconds
          P = AtoP(M_encl, M2, CE_companion_position*Rsun)
 
          ! Determine Keplerian velocity. Then subtract the local rotation velocity
          if (s%rotation_flag) then
             omega_at_companion = s% omega(k) + (CE_companion_position*Rsun - s% r(k)) * &
                  (s% omega(k-1)-s% omega(k)) / (s% r(k-1) - s% r(k))
-            v_rel_at_companion = 2.0 * pi * CE_companion_position*Rsun / P - &
-                 omega_at_companion * CE_companion_position*Rsun ! local rotation velocity = omega * r
+            ! check if the velocity is computed on early steps
+            if (s% x_logical_ctrl(7)) then 
+               v_rel_at_companion = sqrt( (s% xtra(26) * Rsun / secyer)**2.d0 +  &
+                  (s% xtra(27) * s% xtra(2) * Rsun / secyer - omega_at_companion * CE_companion_position*Rsun )**2.d0)
```

Comment:
Aren't `s% xtra(2)` and `CE_companion_position` the same? Hence, why using different names here?